core/dbt/events/functions.py

```diff
@@ -310,13 +309,15 @@ def fire_event(e: Event) -> None:
     # skip logs when `--log-cache-events` is not passed
     if isinstance(e, Cache) and not flags.LOG_CACHE_EVENTS:
         return
+
     # if and only if the event history deque will be completely filled by this event
     # fire warning that old events are now being dropped
     global EVENT_HISTORY
-    if len(EVENT_HISTORY) == ((EVENT_HISTORY.maxlen or 100000) - 1):
+    if len(EVENT_HISTORY) == (flags.EVENT_BUFFER_SIZE - 1):
```

Comment:
@nathaniel-may Sorry I missed this in the first PR feedback, but this pattern is required or fire_event(EventBufferFull()) will create a recursion loop. :|

Reply:
ahh right.